os/include/tinyara/pm/pm.h

```diff
@@ -315,12 +318,16 @@ struct pm_arg_s {
 
 typedef struct pm_arg_s pm_arg_t;
 
-struct pm_timer_s {
-	uint8_t timer_type;		/* Bits here are set according to the timer that is to be used */
-	uint32_t timer_interval;	/* The interval for whichever timer is to be used */
-	clock_t last_wifi_alive_send_time;       /* Time tick of the last wifi keep alive signal sent time */
+struct pm_wakeup_timer_s {	
+	struct pm_wakeup_timer_s *next;   /* pointer to next timer in the linked list */
+	int pid;                          /* id of process that created pm timer */
+	uint8_t flags;                    /* See pm timer definations above*/
+	int delay;		          /* refers to time of sleep expected */
+	sem_t pm_sem;			  /* semaphore to lock the thread that started the pm timer */
```

Comment:
you are providing this as a public API so that pm_wakeup_timer can be used elsewhere besides pm_sleep.

Then pm_sem should not be here.

Add a callback to be called when the timer expires and change it to receive pm_sem as an argument.